containers/pre_built_workers/cxx/Dockerfile

```diff
@@ -0,0 +1,45 @@
+# Copyright 2020 gRPC authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM l.gcr.io/google/bazel:latest
+
+RUN mkdir -p /source/code
+WORKDIR /source/code
+
+# Clone source code into directory
+ARG REPOSITORY=grpc/grpc
+ARG GITREF=master
```

Comment:
is the "GITREF" going to be persisted in the final image so I can later recall its value?

Since we're trying to make the resulting image small and in some cases just storing the "qps_worker" binary will be enough (we don't need to keep the sourcecode etc.), it makes sense to store a small text file in the final image that has the git commit from which we've originally built the binary.
This can be later used for debugging / investigation.

I suggest storing a file like `GRPC_GIT_COMMIT` file somewhere in the final image (next to the qps_worker binary?) and contents of the file would be the commit sha we used to build.

Note that GITREF and commit sha are different and gitref can be "master" and that's pretty useless for figuring out where the binary came from once we no longer have the source code. We should therefore store the commit sha because it's more useful for investigation.

Reply:
I totally agree this information is valuable for debugging. I update the Dockerfile to make the COMMIT SHA accessible in the destination image. The 'COMMIT SHA' is saved in '/source/code/GRPC_GIT_COMMIT.txt'. I am happy to make further modifications needed.

Reply:
Just to add to this discussion, we will need the code for interpreted languages, like ruby. Just not for languages that produce an executable.